src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -63,7 +63,7 @@ public void process(WatchedEvent event) throws Exception {
             notifyListeners(Event.nodeDeleted());
             break;
           default:
-            fetch(false);
+            fetchInExecutor();
```

Comment:
I removed the boolean parameter to `fetch` because the correct value could be lost in the various callbacks. I think we just want to ensure we don't close curator while running in its executor, so I've made that explicit with the `fetchInExecutor` method.

Reply:
The ZooKeeper docs seem to suggest that you won't see the write until the watcher fires:
https://zookeeper.apache.org/doc/r3.1.2/zookeeperProgrammers.html#sc_WatchGuarantees
> A client will see a watch event for a znode it is watching before seeing the new data that corresponds to that znode.

But because they're using the same single-threaded executor I guess it's possible for the watcher event to be queued on the executor behind this task and replace the curator unnecessarily. At least this doesn't cause a problem for correctness as far as I can tell